### ISSUE TITLE
Build an asset font for custom icons

### ISSUE THREAD

**Issue by gnprice** [zulip/zulip-flutter#200]
Much like we do in zulip-mobile and Zulip web. We'll want a font file containing the icons, and then a generated Dart file that maps icon names to glyph IDs in the font.

For references, see:
* `tools/build-icon-font` in zulip-mobile, which is a Node script doing a similar job using the NPM package `@vusion/webfonts-generator`
* [the Dart script](https://github.com/lucide-icons/lucide/blob/34155d48e7927afe80512a4ba4994085e799e52c/packages/lucide-flutter/tool/generate_fonts.dart) that Lucide uses to generate the mapping file when building their Flutter package (but that script consumes a file that something else generated, which provides the data just in a different format)

It would be nice if we can write the script purely in Dart. If we can't readily lay hands on appropriate tools to make that easy, we could settle for having a Node script so that we can use NPM packages.

The generated output — both the font file and the generated Dart file — should be committed to the repo, just like we do for other generated files and just like we do for the corresponding files in zulip-mobile. That way the script only has to be run when changing the set of icons, and there's no complication added to a normal development cycle.

**Comment by gnprice** [comment]
The custom icons we'll want are those [in `@zulip/shared`](https://github.com/zulip/zulip/tree/92c83c1df489975f8fe887f190e5d0af7c9534a2/web/shared/icons), used by Zulip web and zulip-mobile.

Potentially we might also want to include other icons not found there.

One known place where we'll want to use a custom icon is in #119, in the spot that replaces an avatar for group DM threads; see [chat thread](https://chat.zulip.org/#narrow/stream/243-mobile-team/topic/design.3A.20DM-conversation.20list/near/1594671). If we write this script after #119 is already done, then we'll have some other icon there initially and we'll want to replace it with the custom icon.
